dev/run-tests.py

```diff
@@ -0,0 +1,417 @@
+    print "[error] running", cmd, "; received return code", retcode
+    sys.exit(int(os.environ.get("CURRENT_BLOCK", 255)))
+
+
+def rm_r(path):
+    """Given an arbitrary path properly remove it with the correct python
+    construct if it exists
+    - from: http://stackoverflow.com/a/9559881"""
+
+    if os.path.isdir(path):
+        shutil.rmtree(path)
+    elif os.path.exists(path):
+        os.remove(path)
+
+
+def lineno():
+    """Returns the current line number in our program
+    - from: http://stackoverflow.com/a/3056059"""
+
+    return inspect.currentframe().f_back.f_lineno
+
+
+def run_cmd(cmd):
+    """Given a command as a list of arguments will attempt to execute the
+    command and, on failure, print an error message"""
+
+    if not isinstance(cmd, list):
+        cmd = cmd.split()
+    try:
+        subprocess.check_call(cmd)
+    except subprocess.CalledProcessError as e:
+        exit_from_command_with_retcode(e.cmd, e.returncode)
+
+
+def set_sbt_maven_profile_args():
+    """Properly sets the SBT environment variable arguments with additional
+    checks to determine if this is running on an Amplab Jenkins machine"""
+
+    # base environment values for SBT_MAVEN_PROFILE_ARGS_ENV which will be appended on
+    sbt_maven_profile_args_base = ["-Pkinesis-asl"]
+
+    sbt_maven_profile_arg_dict = {
+        "hadoop1.0" : ["-Phadoop-1", "-Dhadoop.version=1.0.4"],
+        "hadoop2.0" : ["-Phadoop-1", "-Dhadoop.version=2.0.0-mr1-cdh4.1.1"],
+        "hadoop2.2" : ["-Pyarn", "-Phadoop-2.2"],
+        "hadoop2.3" : ["-Pyarn", "-Phadoop-2.3", "-Dhadoop.version=2.3.0"],
+    }
+
+    # set the SBT maven build profile argument environment variable and ensure
+    # we build against the right version of Hadoop
+    if os.environ.get("AMPLAB_JENKINS_BUILD_PROFILE"):
+        os.environ[SBT_MAVEN_PROFILE_ARGS_ENV] = \
+            " ".join(sbt_maven_profile_arg_dict.get(ajbp, []) 
+                     + sbt_maven_profile_args_base)
+    else:
+        os.environ[SBT_MAVEN_PROFILE_ARGS_ENV] = \
+            " ".join(sbt_maven_profile_arg_dict.get("hadoop2.3", [])
+                     + sbt_maven_profile_args_base)
+
+
+def is_exe(path):
+    """Check if a given path is an executable file
+    - from: http://stackoverflow.com/a/377028"""
+
+    return os.path.isfile(path) and os.access(path, os.X_OK)
+
+
+def which(program):
+    """Find and return the given program by its absolute path or 'None'
+    - from: http://stackoverflow.com/a/377028"""
+
+    fpath, fname = os.path.split(program)
+
+    if fpath:
+        if is_exe(program):
+            return program
+    else:
+        for path in os.environ.get("PATH").split(os.pathsep):
+            path = path.strip('"')
+            exe_file = os.path.join(path, program)
+            if is_exe(exe_file):
+                return exe_file
+    return None
+
+
+def determine_java_executable():
+    """Will return the *best* path possible for a 'java' executable or `None`"""
```

Comment:
This should actually say it returns the same java executable that will be used by Spark's tests (rather than the "best"). Also it would be good to add an internal comment saying that any changes in the way that Spark's build detects java must be reflected here. Currently the build looks for $JAVA_HOME/bin/java then falls back to the java on the path (which you correctly mimic here).